Create Polkanalysis.md

Apolixit opened the pull request:
#### Project Abstract

Polkanalysis is a Substrate-based blockchain explorer.
It’s a web application and an API that combines data directly retrieved from the chain with a background service that also performs on-chain queries to populate a database.

#### Grant [level](https://github.com/w3f/Grants-Program#level_slider-levels)

- [ ] **Level 1**:  Up to $10,000, 2 approvals
- [x] **Level 2**:  Up to $30,000, 3 approvals
- [ ] **Level 3**:  Unlimited, 5 approvals (for >$100k: Web3 Foundation Council approval)

#### Application Checklist

- [x] The [application template](https://github.com/w3f/Grants-Program/blob/master/applications/application-template.md) has been copied and aptly renamed (`project_name.md`).
- [x] I have read the [application guidelines](https://github.com/w3f/Grants-Program/blob/master/docs/Support%20Docs/grant_guidelines_per_category.md).
- [ ] Payment details have been provided (Polkadot AssetHub (USDC & DOT) address in the application and bank details via email, if applicable).
- [x] I understand that an agreed upon percentage of each milestone will be paid in vested DOT, to the Polkadot address listed in the application.
- [x] I am aware that, in order to receive a grant, I (and the entity I represent) have to successfully complete a KYC/KYB check.
- [x] The software delivered for this grant will be released under an open-source license specified in the application.
- [x] The initial PR contains only one commit (squash and force-push if needed).
- [x] The grant will only be announced once the first milestone [has been accepted](https://github.com/w3f/Grant-Milestone-Delivery#process) (see the [announcement guidelines](https://github.com/w3f/Grants-Program/blob/master/docs/Support%20Docs/announcement-guidelines.md)).
- [ ] I prefer the discussion of this application to take place in a private Element/Matrix channel. My username is: `@_______:matrix.org` (change the homeserver if you use a different one)

Apolixit commented:
I have read and hereby sign the Contributor License Agreement.

semuelle reviewed (commented):
Hi @Apolixit, thank you for the application. I'm impressed with the work you have already put into the project and appreciate that you are moving the .NET tools forward. Two issues:
1. We don't usually fund work retroactively. At the same time, we don't want to punish people who work on projects in their own time. If you could clarify how much time you are still planning to put into the project before submitting the milestone, it could help us better understand where you are in the process. Also, it would help with the decision if the application was more focused on the remaining part. 
2. We have funded a number of explorers and related tools, and almost all of them fail to get a sufficient number of paying customers to keep up operations. If you are - at least initially - not planning to compete with the major explorers, how are you planning to get there?

Lastly, the deliverables listed under `9` are currently published under GPL. The milestone spec lists MIT as the license, so unless you are planning to change the license, I suggest that you remove them. Feel free to list them separately under "Additional Work" or a similar heading.

Apolixit commented:
Hello @semuelle ,
Thank you very much for your feedback!

Here are the answers to your different points:

For the 9 points mentioned in Milestone 1, the only one that hasn’t been fully developed is the web application (I’ve started it, but let’s say it’s at 50%). The other elements are either completely developed or around 80%. At this point, I can, if you like, add a progress status for each of them.
Nevertheless, I still have around 3 months of development and testing left to stabilize the application. To give you some examples, I still have a few bugs related to CheckMetadaHash and the generation of enums in C# (I noticed that some enums have the same index). I also need to develop several front-end components.
In summary:
I’m at 50% for the web application (3 weeks of work remaining)
I still need to stabilize my infrastructure, as I’m having a few issues mapping my data (1 month)
A few bugs related to the database and concurrent calls since everything is asynchronous (1 week)
Deploy everything in production and start populate the production database (2 weeks)

I will update the PR with the progress percentages for everything.

Regarding the second point, I plan to:

- See how much the cloud database would cost
- If cloud is too expensive, choose a data center and set up my own server (expensive to buy, but after that, the costs are fully controlled)
- Monetize the API to reduce my hosting costs
- Use my own funds if necessary.

The advantage of this kind of part-time project is that it doesn’t need to be profitable (though I will avoid letting it cost me $2,000 a month...), and my financial stability doesn’t depend on it.

Regarding the license, how do you know that it is under GPL?
Thanks a lot :)

Apolixit commented:
Hello @semuelle ,

I’ve updated the PR by splitting my “milestone 1” table into two parts:

- **Milestone 1**: what remains to be done
- **Milestone 0**: what has already been completed in the project
- A **Marketing and long term** section to outline various approaches for making the project profitable

I’ve also updated the licenses of my projects [in this commit](https://github.com/Apolixit/Polkanalysis/pull/37/commits/368923f840c48c58f2bb17d7a9cfea81573dc63b), which will be merged into master in a few days.

Regarding the project description, I don’t believe it needs any changes.  
I understand it’s standard not to accept retroactive payment, but I would like to emphasize that the work done so far also demonstrates the level of rigor I can bring to the project.

Let me know if you need anything else :)
Romain

semuelle commented on applications/polkanalysis.md:
Please rename

Apolixit replied:
Hello @semuelle :)
It is done 👍

semuelle reviewed (changes requested):
Hi @Apolixit, thanks for the updates and sorry for the long radio silence. I'm happy to proceed and will share the application with the rest of the committee once the milestone is renamed.

keeganquigley reviewed (approved):
Thanks for your answers @Apolixit happy to also support it.

PieWol commented:
Hey @Apolixit ,
I'm happy to approve this application. 🎉 

It's great to see that you strongly specify the deliverables for each milestone and already started working on this project prior to applying for a grant. I'm looking forward to your user friendly and visuals focussed blockchain explorer :)

Apolixit commented:
Hello everyone :)

Thank you so much for your approval !
I have completed the KYC process.

How can I keep you updated on the milestones ?
Should I create a new PR each time I complete a feature and include the link to the PR in my project ?

Thank you very much,
Romain

keeganquigley commented:
Thanks @Apolixit I can confirm KYC has been approved :)

To answer your question, there is a separate repo for deliveries; when you are ready to submit a delivery, please create a PR here: https://github.com/w3f/Grant-Milestone-Delivery

No need to submit one for every feature, rather, you would submit once after all the deliverables in milestone 1 are completed. The way it is currently structured, we would basically make a one-time payment for the entire balance, upon completion of M1 per the acceptance of the evaluation. This would happen presumably in 4 months when you are finished.

If you anticipate needing to be paid before that date, I would recommend breaking up the features into two milestones, that way we would pay you half in 2 months, and the rest in 4 months, assuming you stick to the timeline. We pay per milestone, so for example if you wanted to be paid every month you'd need to create 4 milestones.

It's entirely up to you how you want to structure it. Keep in mind that 50% of DOT will be paid out with each milestone, which will effectively create separate vesting start dates.

keeganquigley commented:
Hi @Apolixit can you please provide an update on the status of M1?

Apolixit commented:
Hello @keeganquigley,
My apologies — I'm still working on it, but I recently started a new job that's taking up a lot of my time :)

I should be able to deliver M1 by September, or October at the latest. Does that still work for you?
Thanks a lot,
Romain

keeganquigley commented:
Thanks for the update @Apolixit that should work. Normally I'd ask for an amendment to extend the timeline, but since the grant already has a two-year runway I think it's not necessary in this case. If you can deliver everything in the fall I think that should be fine. Thanks for checking!
